website/guides/sgb_border.md

```diff
@@ -0,0 +1,178 @@
+
+## TRN
+
+Bulk transfer (TRN) packets tell the SGB to copy the contents of the screen to buffers in Super NES work RAM. The `CHR_TRN` and `PCT_TRN` packets are used to send data for SGB borders.
+
+For that to function properly, you must:
+
+1. set `BGP` to `$e4` and `LCDC` to `$91` (screen enabled, BG uses tiles `$8000`-`$8fff` and tilemap `$9800`, WIN and OBJ disabled, BG enabled)
+2. set `SCX` and `SCY` to `$00`
+3. the tilemap consists of `$00`, `$01`..`$13`, 12 bytes padding (offscreen), `$14`..`$27`, padding, repeat until `$ff` (inclusive)
+4. the data you want to send must be loaded at `$8000`-`$8fff`
+
+You can do 1, 2 and 3 via [this snippet](https://github.com/zlago/snek-gbc/blob/baef0369f57d2b0d58316cb1c28c6cc22475a6c9/code/init.sm83#L208-L230)
+
+- **You must load the data into VRAM and enable the screen before sending the TRN packet**
+- **You must wait ~8 frames after each TRN** instead of just 4
+
+## Detecting SGB
+
+Here's how a SGB detection routine should look like:
+
+- wait 12 or more frames after the console boots
+- send a `MLT_REQ`, selecting 2 or 4 players (`$89, $01` or `$89, $03`), and wait at least 1 frame
+- attempt to advance the read player (set `P1.5` to 0 then 1)
+- set `P1.4`-`P1.5` to `%11` (`%xx11xxxx`)
+- read `P1.0`-`P1.3`, and check if it either
+	* has changed
+	* isn't `%1111`
+- advance and reread a few times and branch somewhere if the test keeps failing
+
+Let's go over an example snippet ([source](https://github.com/zlago/snek-gbc/blob/baef0369f57d2b0d58316cb1c28c6cc22475a6c9/code/init.sm83#L167-L196)):
+
+```sm83asm
+; test for SGB
+xor a /* first it enables STAT interrupt, since snek-gbc happens to just have a `reti` as the handler */
+ldh [rLYC], a
+ld a, STATF_LYC
+ldh [rSTAT], a
+ld a, IEF_STAT
+ldh [rIE], a
+ei
+	; wait for SGB
+	ld b, 12 /* You must wait 12 or more frames before trying to send a packet */
+	:halt
+	dec b
+	jr nz, :-
+	; enable multi
+	ld hl, Packets.mlt /* send a `MLT_REQ` */
+	call Packet
+	rept 4 /* then wait 4 more frames, since the SNES won't "read" the packet instantly */
+		halt
+	endr
+	; check if SGB responds
+		/* and now We actually try to detect the SGB
+		setting `P1.5` to low then high advances the "read player"
+		setting `P1.4` and `P1.5` high will make the SGB return which player is currently selected in `P1.0` and `P1.1` */
+		lb bc, 5, LOW(rP1) /* b is loaded with 5 which is how many times We try to check if the player changes to anything but P1 */
+		:ld a, P1F_4 /* try to advance player */
+		ldh [c], a
+		ld a, P1F_4|P1F_5 /* then set P1 to return the current player */
+		ldh [c], a
+		ldh a, [c]
+		dec b
+		jp z, .init ; give up /* if 5 (4 actually) attempts fail, assume this isn't an SGB */
+		cp $ff /* `P1.6` and `P1.7` always return %1, We set `P1.4` and `P1.5` to %1, and DMG, or SGB player 1, return %1111 in `P1.0`-`P1.3` */
+		jr z, :- ; /* try again if detection fails */
+```
```

Comment:
`sm83asm` is not recognised by any highlighter out there; I don't know which one is used on gbdev.io, but `rgbasm` is supported by highlight.js [via a plugin](https://github.com/gbdev/highlightjs-rgbasm), and alternatives like `avrasm`, `z80asm`, `x86asm`, or `nasm` degrade well enough.

Also, "block" comments (`/* */`) are very uncommon in assembly syntaxes; single-line comments (`;`) should be used instead, especially if using one of the alternatives listed above.

Reply:
Agree on both points here